web/src/stream_edit.js

```diff
@@ -465,6 +465,34 @@ export function initialize() {
         return true;
     });
 
+    $("#streams_overlay_container").on(
+        "click",
+        ".stream-permissions-warning-banner .main-view-banner-close-button",
+        (event) => {
+            event.preventDefault();
+            $("#stream_permission_settings .stream-permissions-warning-banner").empty();
+        },
+    );
+
+    $("#streams_overlay_container").on(
+        "click",
+        ".stream-permissions-warning-banner .main-view-banner-action-button",
+        (event) => {
+            event.preventDefault();
+            event.stopPropagation();
+
+            const sub = get_sub_for_target(event.target);
+            // Makes sure we take the correct stream_row.
+            const $stream_row = $(
+                `#streams_overlay_container div.stream-row[data-stream-id='${CSS.escape(
+                    sub.stream_id,
+                )}']`,
+            );
```

Comment:
Would it be cleaner if we just had a `data-stream-id` on the "Subscribe" button, rather than needing to pull it out of the global attributes for the panel?

Reply:
Ahh, indeed, `render_compose_banner` definitely accepts  a `stream_id` parameter...

Reply:
```diff

diff --git a/web/src/settings_org.js b/web/src/settings_org.js
index 62f6207b93..7cc1598647 100644
--- a/web/src/settings_org.js
+++ b/web/src/settings_org.js
@@ -1039,10 +1039,11 @@ export function save_discard_widget_status_handler($subsection, for_realm_defaul
             banner_type: compose_banner.WARNING,
             banner_text: $t({
                 defaultMessage:
                    "You are not currently subscribed to this stream. Unless you subscribe, you will lose access after making it private.",
             }),
             button_text: $t({defaultMessage: "Subscribe"}),
             classname: "stream_privacy_warning",
+            stream_id: sub.stream_id,
         };
         $("#stream_permission_settings .stream-permissions-warning-banner").append(
             render_compose_banner(context),
diff --git a/web/src/stream_edit.js b/web/src/stream_edit.js
index e8699bc397..98616fe902 100644
--- a/web/src/stream_edit.js
+++ b/web/src/stream_edit.js
@@ -481,13 +481,15 @@ export function initialize() {
             event.preventDefault();
             event.stopPropagation();
 
-            const sub = get_sub_for_target(event.target);
+            const $target = $(event.target).parents(".main-view-banner");
+            const stream_id = Number.parseInt($target.attr("data-stream-id"), 10);
             // Makes sure we take the correct stream_row.
             const $stream_row = $(
                 `#streams_overlay_container div.stream-row[data-stream-id='${CSS.escape(
-                    sub.stream_id,
+                    stream_id,
                 )}']`,
             );
+            const sub = sub_store.get(stream_id);
             stream_settings_ui.sub_or_unsub(sub, $stream_row);
             $("#stream_permission_settings .stream-permissions-warning-banner").empty();
         },
```